xwiki-rendering-api/src/main/java/org/xwiki/rendering/block/AbstractMacroBlock.java

```diff
@@ -95,21 +95,17 @@ public boolean equals(Object obj)
         if (this == obj) {
             return true;
         }
-
-        if (!super.equals(obj)) {
+        
+        if (!super.equals(obj) || !(obj instanceof AbstractMacroBlock)) {
```

Comment:
Wouldn't that make equals not symmetric. For example, if you were going to create a new subclass of AbstractMacroBlock that required equals to check if some fields are the same or not.(https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#equals(java.lang.Object)).